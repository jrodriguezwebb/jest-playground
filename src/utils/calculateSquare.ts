/* eslint-disable @typescript-eslint/no-explicit-any */
// eslint-disable-next-line @typescript-eslint/ban-types
export function calculateSquareAsync(number: any, callback: Function) {
    setTimeout(() => {
        if(typeof number !== 'number') {
            callback(new Error("Argument of type is expected"));
            return;
        }

        const result = number * number;
        callback(null, result);
    }, 200);
}

export function calculateSquare(number?: any) {
    if(number === undefined || typeof number !== 'number') {
        throw new Error('You must provide a number');
    }
    return number * number;
}
