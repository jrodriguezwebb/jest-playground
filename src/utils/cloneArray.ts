/* eslint-disable @typescript-eslint/no-explicit-any */
export function cloneArray(array: any[]) {
    return [ ...array ];
}
