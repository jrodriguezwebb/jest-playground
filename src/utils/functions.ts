/* eslint-disable @typescript-eslint/no-explicit-any */

export const functions = {
    sum: (a: number, b: number) => {
        return a + b;
    },
    sub: (a: number, b: number) => {
        return a - b;
    },
    copyArray: (array: any[]) => {
        return [ ...array ];
    }
}
