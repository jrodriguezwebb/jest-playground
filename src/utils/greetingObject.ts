export const grettingObject = {
    name: 'Jesus',
    lastName: 'Rodriguez',
    sayHello: function() {
        return `Hello ${ this.name } ${ this.lastName }`;
    }
}