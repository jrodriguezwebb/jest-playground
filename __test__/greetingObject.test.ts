import { grettingObject } from '../src/utils/greetingObject';

test('Expects result of the summarized numbers', () => {
    const helloMessage = grettingObject.sayHello();
    expect(helloMessage).toMatchSnapshot();
})