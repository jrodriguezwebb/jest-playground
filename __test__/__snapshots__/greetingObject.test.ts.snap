// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expects result of the summarized numbers 1`] = `"Hello Jesus Rodriguez"`;
